TO Go: inconsistencies between related structs

in `lib/go-tc`,  there exist `Nullable` versions of most Traffic Ops structs.   There are inconsistencies between these structs that should be addressed.   I've written a test that should be run with unit tests to ensure that these are correct.    I'll create a PR shortly with that test with instructions on how to run it.

The following is the output from the test showing these inconsistencies:
```
go test -tags nullable
--- FAIL: TestNullStructs (0.00s)
    nullable_test.go:97: tc.ASN.LastUpdated (string) and tc.ASNNullable.LastUpdated (*tc.TimeNoMod) have mismatched types
    nullable_test.go:72: tc.DeliveryServiceV11 has 54 fields, but tc.DeliveryServiceNullableV11 has 56
    nullable_test.go:85: field DeepCachingType found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:85: field DNSBypassCname found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:97: tc.DeliveryServiceV11.InitialDispersion (float32) and tc.DeliveryServiceNullableV11.InitialDispersion (*int) have mismatched types
    nullable_test.go:85: field TRResponseHeaders found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:113: field DNSBypassCNAME found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:113: field GeoLimitCountries found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:113: field GeoLimitRedirectURL found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:113: field OriginShield found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:113: field SSLKeyVersion found on tc.DeliveryServiceV11 but not tc.DeliveryServiceNullableV11
    nullable_test.go:72: tc.Profile has 10 fields, but tc.ProfileNullable has 9
    nullable_test.go:85: field Parameter found on tc.Profile but not tc.ProfileNullable
    nullable_test.go:97: tc.SteeringTarget.DeliveryServiceID (int) and tc.SteeringTargetNullable.DeliveryServiceID (*uint64) have mismatched types
    nullable_test.go:97: tc.SteeringTarget.TargetID (int) and tc.SteeringTargetNullable.TargetID (*uint64) have mismatched types
    nullable_test.go:97: tc.SteeringTarget.Value (int) and tc.SteeringTargetNullable.Value (*uint64) have mismatched types
FAIL
exit status 1
FAIL	github.com/apache/trafficcontrol/lib/go-tc	0.045s
```

--- a/lib/go-tc/nullable_test.go
+++ b/lib/go-tc/nullable_test.go
@@ -1,0 +1,115 @@
+// TODO: This comment with the `build nullable` tag must be removed when the structs are made consistent!!
+// skip this unless specifically testing for nullable vs non-nullable struct comparison
+
+// Run as `go test -tags nullable`
+// +build nullable
+
+package tc
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestNullStructs(t *testing.T) {
+	compareWithNullable(t, ASN{}, ASNNullable{})
+	compareWithNullable(t, CacheGroupFallback{}, CacheGroupFallbackNullable{})
+	compareWithNullable(t, CacheGroup{}, CacheGroupNullable{})
+	compareWithNullable(t, CDN{}, CDNNullable{})
+	compareWithNullable(t, Coordinate{}, CoordinateNullable{})
+	compareWithNullable(t, DeliveryServiceRequestComment{}, DeliveryServiceRequestCommentNullable{})
+	compareWithNullable(t, DeliveryServiceRequest{}, DeliveryServiceRequestNullable{})
+	compareWithNullable(t, DeliveryService{}, DeliveryServiceNullable{})
+	compareWithNullable(t, DeliveryServiceV12{}, DeliveryServiceNullableV12{})
+	compareWithNullable(t, DeliveryServiceV11{}, DeliveryServiceNullableV11{})
+	compareWithNullable(t, Division{}, DivisionNullable{})
+	compareWithNullable(t, Domain{}, DomainNullable{})
+	compareWithNullable(t, Parameter{}, ParameterNullable{})
+	compareWithNullable(t, PhysLocation{}, PhysLocationNullable{})
+	compareWithNullable(t, ProfileParameter{}, ProfileParameterNullable{})
+	compareWithNullable(t, Profile{}, ProfileNullable{})
+	compareWithNullable(t, Server{}, ServerNullable{})
+	compareWithNullable(t, StaticDNSEntry{}, StaticDNSEntryNullable{})
+	compareWithNullable(t, Status{}, StatusNullable{})
+	compareWithNullable(t, SteeringTarget{}, SteeringTargetNullable{})
+	compareWithNullable(t, Tenant{}, TenantNullable{})
+	compareWithNullable(t, Type{}, TypeNullable{})
+
+	// No Nullable version of these types
+	//compareWithNullable(t, Federation{}, FederationNullable{})
+	//compareWithNullable(t, ProfileParameters{}, ProfileParametersNullable{})
+}
+
+// compareFields checks that non-nullable and nullable versions have same fields
+func compareWithNullable(t *testing.T, obj interface{}, nullObj interface{}) {
+	ot := reflect.TypeOf(obj)
+	nt := reflect.TypeOf(nullObj)
+	if strings.Replace(nt.String(), "Nullable", "", 1) != ot.String() {
+		t.Errorf("expected type %s with nullable %s", ot, nt)
+	}
+
+	if ot.NumField() != nt.NumField() {
+		t.Errorf("%T has %d fields, but %T has %d", obj, ot.NumField(), nullObj, nt.NumField())
+	}
+
+	seen := make(map[string]struct{}, ot.NumField())
+
+	for i := 0; i < ot.NumField(); i++ {
+		oField := ot.Field(i)
+		if oField.Anonymous {
+			// embedded struct -- skip it
+			continue
+		}
+		nField, ok := nt.FieldByName(oField.Name)
+		if !ok {
+			t.Errorf("field %s found on %T but not %T", oField.Name, obj, nullObj)
+			continue
+		}
+
+		seen[nField.Name] = struct{}{}
+		oKind := oField.Type.Kind()
+		nKind := nField.Type.Kind()
+		//t.Logf("%T.%s is %s. %T.%s is %s", obj, oField.Name, oKind.String(), nullObj, nField.Name, nKind.String())
+		if oKind == nKind {
+			continue
+		}
+		if nKind == reflect.Ptr && nField.Type.Elem().Kind() != oKind {
+			t.Errorf("%T.%s (%s) and %T.%s (%s) have mismatched types", obj, oField.Name, oField.Type.String(), nullObj, nField.Name, nField.Type.String())
+		}
+	}
+
+	// check for fields in Nullable version not in non-Nullable
+	for i := 0; i < nt.NumField(); i++ {
+		nField := nt.Field(i)
+		if nField.Anonymous {
+			// embedded struct -- skip it
+			continue
+		}
+		if _, ok := seen[nField.Name]; ok {
+			// already accounted for
+			continue
+		}
+
+		t.Errorf("field %s found on %T but not %T", nField.Name, obj, nullObj)
+	}
+}

